### PR TITLE
Flex fixes

### PR DESCRIPTION
**Pull request by badpandabear**
Moved call to detect_lines on the archived sky spectrum to spec_flexure_obj for performance.

Also has fixes from @profxj for an intermittent flexure failure I was seeing.
Unit Tests:

> ==================================================================== 235 passed, 562 warnings in 365.86s (0:06:05) ====================================================================

Dev suite:

> --- PYPEIT DEVELOPMENT SUITE FAILED 1/75 TESTS (Masters ignored) ---
> Failed tests:
>     pypeit_ql: keck_nires/NIRES
> 

(I had accidentally synced over NIRES_MASTERS)

> Running tests on the following instruments:
>     keck_nires
> 
> Reducing data from keck_nires for the following setups:
>     NIRES
> 
> Directories:
>          Raw data: /home/dusty/work/PypeIt-development-suite/RAW_DATA/keck_nires/NIRES
>     PypeIt output: /home/dusty/work/PypeIt-development-suite/REDUX_OUT/keck_nires/NIRES
> 
> Directory: /home/dusty/work/PypeIt-development-suite/REDUX_OUT/keck_nires/NIRES
> Command line: run_pypeit keck_nires_nires.pypeit -o
> Wrote: /home/dusty/work/PypeIt-development-suite/REDUX_OUT/keck_nires/NIRES/QA/MF_A.html
> Running pypeit on keck_nires/NIRES (ignores masters)  --- PASSED
> [INFO]    :: Data reduction complete
> [INFO]    :: Generating QA HTML
> 
> 
> Directory: /home/dusty/work/PypeIt-development-suite/REDUX_OUT/keck_nires/NIRES/QL
> Command line: pypeit_ql_keck_nires /home/dusty/work/PypeIt-development-suite/RAW_DATA/keck_nires/NIRES s190519_0067.fits s190519_0068.fits
> Wrote: /home/dusty/work/PypeIt-development-suite/REDUX_OUT/keck_nires/NIRES/QL/keck_nires_A/QA/MF_A.html
> Running pypeit_ql_mos on keck_niresQL  --- PASSED
> [INFO]    :: Execution time: 37.47s
> [INFO]    :: Data reduction complete
> [INFO]    :: Generating QA HTML
> 
> 
> 
> --- PYPEIT DEVELOPMENT SUITE PASSED 2/2 TESTS (Masters ignored) ---
>

### CODE REVIEW

File: pypeit/core/flexure.py [hunk 1]
```diff
@@ -120,14 +120,16 @@ def load_sky_spectrum(sky_file):
...
-def spec_flex_shift(obj_skyspec, arx_skyspec, mxshft=20):
+def spec_flex_shift(obj_skyspec, arx_skyspec, arx_lines, mxshft=20):
     """ Calculate shift between object sky spectrum and archive sky spectrum
 
     Args:
         obj_skyspec (:class:`linetools.spectra.xspectrum1d.XSpectrum1d`):
             Spectrum of the sky related to our object
         arx_skyspec (:class:`linetools.spectra.xspectrum1d.XSpectrum1d`):
             Archived sky spectrum
+        arx_lines (list): Line information returned by arc.detect_lines for
```
Comment (rcooke-ast) [inline pypeit/core/flexure.py:131]: I think this will technically be a `tuple` instead of a `list`?

**Comment by badpandabear** [comment]
> Looks great, thanks.
> 
> Can you also add a comment or two on the approximate speed-up for
> DEIMOS from the change? Just for posterity. Thanks.

To clarify: Comment in the code or on the pull request?

**Comment by profxj** [comment]
Here in the pull request is good.
Thanks.

**Comment by badpandabear** [comment]
Time for the keck_deimos 1200G_M_7750 test case, all detectors and images before these changes:  266m 25s
Time after: 230m 29s
A performance improvement of about 36 minutes or 13% of the total runtime.

**Comment by profxj** [comment]
Great, thanks!
merging.

**Pull request merged**